manifold/resource_manifold_token.go

```diff
@@ -0,0 +1,125 @@
+package manifold
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	manifold "github.com/manifoldco/go-manifold"
+	"github.com/manifoldco/kubernetes-credentials/helpers/client"
+)
+
+func resourceManifoldToken() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceManifoldTokenCreate,
+		Read:   resourceManifoldTokenRead,
+		Update: resourceManifoldTokenUpdate,
+		Delete: resourceManifoldTokenDelete,
+
+		Schema: map[string]*schema.Schema{
+			"role": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: "The role this key should assume.",
+			},
+
+			"self": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Assign the token to user that is linked to the configuration token.",
+			},
```

Comment:
should you be able to specify a team, too? Like, i could use my personal token to connect to terraform, and create a read-credentials token for a team i belong to, and pop that token into a k8s cluster